### PR TITLE
Update zha.markdown to mention and explain binding and unbinding

### PR DESCRIPTION
**Pull request by Hedda**
## Proposed change

Added missing information to current documentation in zha.markdown about Zigbee binding and unbinding for remotes, etc.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: 

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

**Comment by Hedda** [comment]
> It seems that this PR is targeted against an incorrect branch since it has a parent PR on one of our codebases. Documentation that needs to be updated for an upcoming release should target the `next` branch. Please change the target branch of this PR to `next` and rebase if needed.
> (message by DocsTargetBranch)

hmm, no, this PR is correctly targeting the current branch on purpose. Unbinding is supported since https://github.com/home-assistant/core/pull/42854 PR.

**Comment by Hedda** [comment]
Is it possible to remove tags for "has-parent" and "needs-rebase" from this pull request as those are not relevant to this PR?

Binding existed before and Zigbee coordinator unbinding was added in https://github.com/home-assistant/core/pull/42854 back in November 2020.

**Comment by Hedda** [comment]
Bump! This should not need rebasing? Should it?

**Review by jampez77 — DISMISSED** [review]

**Comment by Hedda** [comment]
Is it possible to remove "needs-rebase" and "has-parent" labels/tags from this pull request as those are incorrectly labelled/tagged?

**Comment by frenck** [comment]
> Is it possible to remove "needs-rebase" and "has-parent" labels/tags from this pull request as those are incorrectly labelled/tagged?

Will be done when this PR is reviewed. No need to comment on that. Thanks 👍

File: source/_integrations/zha.markdown [hunk 1]
```diff
@@ -273,7 +273,7 @@ ZHA support for binding and unbinding. Binding is an action in Zigbee which defi
...
 
-By default ZHA bind remotes to the coordinator, so coordinator could receive ZCL commands from the remotes and originate zha_event's. However some remotes, like example Philips RWL021 can only be bound to a single destination and it is not possible to make this switch to bind to another destinations like a device or groups unless you first unbind the remote from the coordinator. After you unbind the remote from ZHA coordinator you can then bind it directly to any other Zigbee device or a group directly.
+Note that all devices support binding as it depends on the Zigbee implementation of the device itself. Also, by default ZHA bind remotes to the coordinator, so coordinator could receive ZCL commands from the remotes and originate zha_event's. However some remotes, like example Philips RWL021 can only be bound to a single destination and it is not possible to make this switch to bind to another destinations like a device or groups unless you first unbind the remote from the coordinator. After you unbind the remote from ZHA coordinator you can then bind it directly to any other Zigbee device or a group directly.
```
Comment (reedtaylor) [inline source/_integrations/zha.markdown]: I think this may be missing the word "not" and is in fact saying the opposite of the intended meaning.

Comment (Hedda) [reply source/_integrations/zha.markdown]: Indeed. Fixed now.

**Comment by Hedda** [comment]
Does anyone know what is needed to fix that lint text test check? I do not see the misspelt word that the check refers to.

Maybe this ZHA docs PR should be reviewed by @dmulcahey or @Adminiuga regardless as they are maintainers of ZHA?

**Comment by frenck** [comment]
```
/home/runner/work/home-assistant.io/home-assistant.io/source/_integrations/zha.markdown
  340:54  ✓ error  This is a commonly misspelled word. Correct it to benefit  common-misspellings
```

Additionally, @Hedda Please don't ping maintainers. The bot takes care of that. It is not allowed and/or appreciated to do so. Thanks 👍

File: source/_integrations/zha.markdown [hunk 2]
```diff
@@ -267,6 +267,16 @@ An example using the default CC2531 coordinator firmware + two CC2531 routers; Y
...
+
+Note that not all devices support binding as it depends on the Zigbee implementation of the device itself. Also, by default ZHA bind remotes to the coordinator, so coordinator could receive ZCL commands from the remotes and originate zha_events. However, some remotes, like example the Philips RWL021 can only be bound to a single destination and it is not possible to make this switch to bind to another destinations like a device or groups unless you first unbind the remote from the coordinator. After you unbind the remote from ZHA coordinator you can then bind it directly to any other Zigbee device or a group.
+
+Binding a remote directly to a bulb or group has the benifit of faster response time and smoother control. This benefit use greatly improve user feedback experience functions like dimming as the remote then directly dims the lightbulb and thus does not have to make the software roundtrip via the ZHA coordinator.
```
Comment (frenck) [inline source/_integrations/zha.markdown]: There you go.
```suggestion
Binding a remote directly to a bulb or group has the benefit of faster response time and smoother control. This greatly improves user feedback experience functions like dimming as the remote then directly dims the lightbulb and thus does not have to make the software roundtrip via the ZHA coordinator.
```

**Comment by PDekker** [comment]
@frenck Just some feedback from the enduser. I has to lean this from this open pull request from January.

`Note that all devices support binding as it depends on the Zigbee implementation of the device itself. Also, by default ZHA bind remotes to the coordinator, so coordinator could receive ZCL commands from the remotes and originate zha_event's. However some remotes, like example Philips RWL021 can only be bound to a single destination and it is not possible to make this switch to bind to another destinations like a device or groups unless you first unbind the remote from the coordinator. After you unbind the remote from ZHA coordinator you can then bind it directly to any other Zigbee device or a group directly.`

You guys can do better than this!

File: source/_integrations/zha.markdown [hunk 3]
```diff
@@ -267,6 +267,16 @@ An example using the default CC2531 coordinator firmware + two CC2531 routers; Y
 - Router two: + 21 devices
 - Total device limit = **55 devices**
 
+### Binding and unbinding
+
+ZHA support for binding and unbinding. Binding is an action in Zigbee which defines relations between two Zigbee devices, specific endpoints and cluster id. It provides a mechanism for attaching an endpoint on one Zigbee node to one or more endpoints on another Zigbee node and can even be destined for Zigbee groups of nodes.
```
Comment (Adminiuga) [inline source/_integrations/zha.markdown]:
```suggestion
ZHA support for binding and unbinding. Binding is an action in Zigbee which defines relations between two Zigbee devices, specific endpoints and cluster id. It provides a mechanism for attaching an endpoint on one Zigbee node to one or more endpoints on another Zigbee node or Zigbee group (a group of Zigbee devices). 
```

File: source/_integrations/zha.markdown [hunk 4]
```diff
@@ -267,6 +267,16 @@ An example using the default CC2531 coordinator firmware + two CC2531 routers; Y
...
 - Total device limit = **55 devices**
 
+### Binding and unbinding
+
+ZHA support for binding and unbinding. Binding is an action in Zigbee which defines relations between two Zigbee devices, specific endpoints and cluster id. It provides a mechanism for attaching an endpoint on one Zigbee node to one or more endpoints on another Zigbee node and can even be destined for Zigbee groups of nodes. 
+
+Binding of Zigbee clusters is nothing but establishing direct virtual connections between endpoints on different Zigbee nodes so that, during data transmission between the connected endpoints, the destination is identified by a specific cluster, rather than by the destination address. This is useful to for example bind a Zigbee end device like a remote or motion-sensor to a Zigbee lightbulb or group of lightbulbs. After this binding is done the remote or motion-sensor is attached directly to that device via a peer-to-peer connection and can direcly control that bound lightbulb / group of lightbulbs without going through the Zigbee coordinator. This means that the remote or motion-sensor can control the lightbulb  / group of lightbulbs even when the Zigbee coordinator is not available.
```
Comment (Adminiuga) [inline source/_integrations/zha.markdown]:
```suggestion
Binding is a "target destination" in form of a device address or group ID, endpoint and cluster. For example binding a Zigbee device like a remote to a Zigbee lightbulb, switch or group of lightbulbs allows direct control of the "target" device (light, switch, shade) from the "remote" Zigbee device, bypassing ZHA.  This means that the remote can control the lightbulb  / group of lightbulbs even when the Zigbee coordinator is not available.
Binding is only supported between the same cluster, for example "output cluster id 6" (on/off cluster) of a remote, can be only bound to an "input cluster id 6" on the target device -- light, switch.  
```
 
Why such long lines???

File: source/_integrations/zha.markdown [hunk 5]
```diff
@@ -267,6 +267,17 @@ An example using the default CC2531 coordinator firmware + two CC2531 routers; Y
...
+Binding is only supported between the same cluster, for example "output cluster id 6" (on/off cluster) of a remote, can be only bound to an "input cluster id 6" on the target device -- light, switch.  
+
+Note that not all devices support binding as it depends on the Zigbee implementation of the device itself. Also, by default ZHA bind remotes to the coordinator, so coordinator could receive ZCL commands from the remotes and originate zha_events. However, some remotes, like example the Philips RWL021 can only be bound to a single destination and it is not possible to make this switch to bind to another destinations like a device or groups unless you first unbind the remote from the coordinator. After you unbind the remote from ZHA coordinator you can then bind it directly to any other Zigbee device or a group.
+
+Binding a remote directly to a bulb or group has the benefit of faster response time and smoother control. This greatly improves user feedback experience functions like dimming as the remote then directly dims the lightbulb and thus does not have to make the software roundtrip via the ZHA coordinator.
```
Comment (frenck) [inline source/_integrations/zha.markdown]:
```suggestion
### Binding and unbinding

ZHA support for binding and unbinding. Binding is an action in Zigbee which defines relations between two Zigbee devices, specific endpoints, and cluster id. It provides a mechanism for attaching an endpoint on one Zigbee node to one or more endpoints on another Zigbee node or Zigbee group (a group of Zigbee devices). 

Binding is a "target destination" in form of a device address or group ID, endpoint, and cluster. For example, binding a Zigbee device like a remote to a Zigbee lightbulb, switch or group of lightbulbs allows direct control of the "target" device (light, switch, shade) from the "remote" Zigbee device, bypassing ZHA.  This means that the remote can control the lightbulb/group of lightbulbs even when the Zigbee coordinator is not available.
Binding is only supported between the same cluster, for example, "output cluster id 6" (on/off cluster) of a remote, can be only bound to an "input cluster id 6" on the target device -- light, switch.  

Note that not all devices support binding as it depends on the Zigbee implementation of the device itself. Also, by default ZHA bind remotes to the coordinator, so the coordinator could receive ZCL commands from the remotes and originate zha_events. However, some remotes, for example, the Philips RWL021 can only be bound to a single destination and it is not possible to make this switch to bind to other destinations like a device or groups unless you first unbind the remote from the coordinator. After you unbind the remote from the ZHA coordinator you can then bind it directly to any other Zigbee device or a group.

Binding a remote directly to a bulb or group has the benefit of faster response time and smoother control. This greatly improves user feedback experience functions like dimming as the remote then directly dims the lightbulb and thus does not have to make the software roundtrip via the ZHA coordinator.
```

**Review by frenck — APPROVED** [review]

**Pull request merged**